public/app/features/alerting/unified/components/rules/RuleDetailsActionButtons.tsx

```diff
@@ -263,6 +267,25 @@ export const RuleDetailsActionButtons: FC<Props> = ({ rule, rulesSource, isViewM
   return null;
 };
 
+/**
+ * We don't want to show the silence button if either
+ * 1. the user has no permissions to create silences
+ * 2. the admin has configured to only send instances to external AMs
+ */
+function useCanSilence(alertmanagerSourceName?: string) {
+  const hasAlertmanagerSource = Boolean(alertmanagerSourceName);
+
+  const { useGetAlertmanagerChoiceStatusQuery } = alertmanagerApi;
+  const { currentData: amConfigStatus } = useGetAlertmanagerChoiceStatusQuery();
+
+  const hasPermissions = contextSrv.hasAccess(AccessControlAction.AlertingInstanceCreate, contextSrv.isEditor);
```

Comment:
This permission is correct only for Grafana AM. 
Please use the `getInstancesPermissions` function to get the necessary permission based on the `alertmanagerSourceName`

Reply:
However, this might be correct because we want to display Silence only for Grafana-managed rules

Reply:
I've copied this one from the original implementation, I believe it's correct since we only show the silence button for Grafana managed rules anyway.